src/document.c

```diff
@@ -473,8 +491,8 @@ FIELD_PREPROCESSOR(geoPreprocessor) {
 
   char *end1 = NULL, *end2 = NULL;
   double lon = strtod(c, &end1);
-  double lat = strtod(pos, &end2);
-  if (*end1 || *end2) {
+  double lat = strtod(pos, &end2); /// failing here
+  if (*end1 || (*end2 && *end2 != '"')) {
```

Comment:
```suggestion
  if (*end1 || (*end2 && (*end2 != '"' || *(end2+1)))) {
```

Reply:
This should reject `"1,2"foo`  (by making sure that quote is the last char)
(BTW, currently input such as `"1,2`  and `1,2"` is accepted)